app/src/androidTest/java/com/github/epfl/meili/models/ChatMessageUnitTest.kt

```diff
@@ -0,0 +1,41 @@
+package com.github.epfl.meili.models
+
+import android.os.Parcel
+import android.util.Log
+import com.github.epfl.meili.models.ChatMessage
+import com.google.firebase.firestore.DocumentSnapshot
+import org.hamcrest.CoreMatchers.`is`
+import org.junit.Test
+
+import org.junit.Assert.*
+
+
+class ChatMessageUnitTest {
+    private val TEST_ID: String = "1"
+    private val TEST_TEXT: String = "Hi!"
+    private val TEST_FROMID: String = "frkhjf243htiu43iut"
+    private val TEST_TOID: String = "fuh43koh31pjh1t4"
+    private val TEST_TIMESTAMP: Long = 1L
+
+
+    @Test
+    fun chatMessageConstructor() {
+        var chatMessage = ChatMessage(TEST_TEXT, TEST_FROMID, TEST_TOID, TEST_TIMESTAMP)
+        assertThat(chatMessage.text, `is`(TEST_TEXT))
+        assertThat(chatMessage.component1(), `is`(TEST_TEXT))
+        assertThat(chatMessage.fromId, `is`(TEST_FROMID))
+        assertThat(chatMessage.component2(), `is`(TEST_FROMID))
+        assertThat(chatMessage.toId, `is`(TEST_TOID))
+        assertThat(chatMessage.component3(), `is`(TEST_TOID))
+        assertThat(chatMessage.timestamp, `is`(TEST_TIMESTAMP))
+        assertThat(chatMessage.component4(), `is`(TEST_TIMESTAMP))
+        assertThat(chatMessage.hashCode(), `is`(-1607505383))
```

Comment:
I would test the equality of hash codes of two message objects created with the same attributes. this is a magic number and we have no idea what it means and whether it's correct

Reply:
The latest commits doens't have this